webui/src/pages/repositories/repository/fileRenderers/index.tsx

```diff
@@ -1,96 +1,342 @@
+                    <DatabaseIcon/>{' '}
+                    Execute
+                </Button>
+            </Form>
+            <div className={"mt-3 object-viewer-sql-results"}>
+                {content}
+            </div>
+        </div>
+    )
+}
+
+interface RowProps {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    value: any;
+}
+
+const DataRow: FC<RowProps> = ({ value }) => {
+     let dataType = 'regular';
+     if (typeof value === 'string') {
+         dataType = 'string';
+     } else if (value instanceof Date) {
+         dataType = 'date'
+     } else if (typeof value === 'number') {
+         dataType = 'number'
+     }
+
+     if (dataType === 'string') {
+         return <td>{value}</td>
+     }
+
+     if (dataType === 'date') {
+         return <td>{moment(value).format(moment.defaultFormat)}</td>
+     }
+
+    if (dataType === 'number') {
+        return <td>{value.toLocaleString("en-US")}</td>
     }
-    return null;
+
+    return <td>{""  + value}</td>;
 }
 
-export const supportedFileExtensionRenderers: Record<string, FC<RendererComponent>> = {
-    md: MarkdownRenderer,
-    ipynb: IpynbRenderer,
-    png: ImageRenderer,
-    jpg: ImageRenderer,
-    jpeg: ImageRenderer,
+export const IpynbRenderer: FC<RendererComponentWithText> = ({ text }) => {
+    return (
+        <NbRenderer
+            ipynb={JSON.parse(text)}
+            syntaxTheme="ghcolors"
+            language="python"
+            bgTransparent={true}
+        />
+    );
 };
 
-export const supportedContentTypeRenderers: Record<string, FC<RendererComponent>> = {
-    "application/x-ipynb+json": IpynbRenderer,
-    "text/markdown":MarkdownRenderer,
-    "image/png": ImageRenderer,
-    "image/jpeg": ImageRenderer,
+const qs = (queryParts: {[key: string]: string}) => {
+    const parts = Object.keys(queryParts).map(key => [key, queryParts[key]]);
+    return new URLSearchParams(parts).toString();
 };
 
+export const ImageRenderer: FC<RendererComponent> = ({ repoId, refId,  path }) => {
+    const query = qs({path});
+    return (
+        <p className="image-container">
+            <img src={`/api/v1/repositories/${encodeURIComponent(repoId)}/refs/${encodeURIComponent(refId)}/objects?${query}`} alt={path}/>
+        </p>
+    );
+}
+
+export const PDFRenderer: FC<RendererComponent> = ({ repoId, refId, path }) => {
+    const query = qs({path});
+    return (
+        <div className="m-3 object-viewer-pdf">
+            <object
+                data={`/api/v1/repositories/${encodeURIComponent(repoId)}/refs/${encodeURIComponent(refId)}/objects?${query}`}
+                type="application/pdf">
+            </object>
+        </div>
+    )
+}
+
+
+
 export const guessLanguage =  (extension: string | null, contentType: string | null) => {
-    if (extension  && SyntaxHighlighter.supportedLanguages.indexOf(extension) !== -1) {
+    if (extension && SyntaxHighlighter.supportedLanguages.indexOf(extension) !== -1) {
```

Comment:
You can use [Array.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) instead of `indexOf`
It's the more modern approach.

Reply:
Nice! Done.